app/scripts/models/account.js

```diff
@@ -394,6 +397,16 @@ define(function (require, exports, module) {
       .then((updatedSessionData) => {
```

Comment:
FWIW you could also use the call to `recoveryEmailStatus` above as an opportunity to make sure the email you're using matches the one held by the server, as it returns the canonical email in its response.  This would help put users who are already signed in into the correct state.